src/python/zquantum/core/interfaces/optimizer_test.py

```diff
@@ -22,3 +25,16 @@ def test_optimization(self):
             self.assertIn("opt_params", results.keys())
             self.assertIn("history", results.keys())
 
+    def test_optimization_simple_function(self):
+        for optimizer in self.optimizers:
+            results = optimizer.minimize(simple, initial_params=[1, -1])
```

Comment:
I think it should not work :) 
Simple is a python function, not `CostFunction` object and therefore it will fail.
Also, actually `MockCostFunction` implements exactly this.

Reply:
Addressed here: 
ec150a4 

Not using the `MockCostFunction` because in theory, we don't know exactly what that cost function is actually doing and here we are making sure it's finding the correct parameters for a known problem.

Reply:
Makes sense 👍